packages/core/src/exposed-thing.ts

```diff
@@ -165,16 +165,20 @@ export default class ExposedThing extends TD.Thing implements WoT.ExposedThing {
                     reject(new Error(`ExposedThing '${this.title}', property '${propertyName}' is writeOnly`));
                 }
 
+                let ps: PropertyState = this.properties[propertyName].getState();
                 // call read handler (if any)
-                if (this.properties[propertyName].getState().readHandler != null) {
+                if (ps.readHandler != null) {
                     console.debug("[core/exposed-thing]",`ExposedThing '${this.title}' calls registered readHandler for Property '${propertyName}'`);
-                    let ps: PropertyState = this.properties[propertyName].getState();
-                    ps.readHandler(options).then((customValue) => {
-                        resolve(customValue);
-                    });
+                    ps.readHandler(options)
+                        .then((customValue) => {
+                            resolve(customValue);
+                        })
+                        .catch((err) => {
+                            reject(err);
+                        });
```

Comment:
Note: This is the actual fix

Reply:
FYI: the changes are very minimal (see https://github.com/eclipse/thingweb.node-wot/pull/341/files?diff=split&w=1)

* introduced variables like ps to avoid writing this.properties[propertyName].getState() over an over again (I should/could have done that in another PR)
* the actual fix is in the 3 lines (see comment above)
* added test-cases (just one fails before this fix "should reject if property read handler fails")